kong/plugins/cors/handler.lua

```diff
@@ -56,8 +58,24 @@ end
 
 local function configure_credentials(ngx, conf)
   if ngx.ctx.cors_allow_all then
-    ngx.header["Access-Control-Allow-Credentials"] = "false"
+    local allow_credentials = false
+
+    -- If it's catch all, then match the Origin requested by the client
+    -- but only if the Origin URL is valid, and only if credentials=true
+    if conf.credentials then
+      local origin = req_get_headers()["Origin"]
+      if origin then
+        local parsed_url = url.parse(origin)
+        if parsed_url.scheme and parsed_url.host then
+          ngx.header["Access-Control-Allow-Origin"] = fmt("%s://%s", 
```

Comment:
@Tieske this applies to `ngx.req.get_headers`, not to settings headers via the `ngx.header` API. Consider the following:

```
server {
  listen 12345
  location /bar {
    content_by_lua 'ngx.say("hello world")';
    header_filter_by_lua 'ngx.header["foo"] = "bar"';
  }
}
```

This produces the following result:

```
$ curl -vvv localhost:12345/bar
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 12345 (#0)
> GET /bar HTTP/1.1
> User-Agent: curl/7.35.0
> Host: localhost:12345
> Accept: */*
> 
< HTTP/1.1 200 OK
* Server openresty/1.11.2.2 is not blacklisted
< Server: openresty/1.11.2.2
< Date: Fri, 24 Mar 2017 16:54:07 GMT
< Content-Type: application/octet-stream
< Transfer-Encoding: chunked
< Connection: keep-alive
< foo: bar
< 
hello world
```

Assigning the header key as camel case is appropriate here.